stl/inc/format

```diff
@@ -366,58 +366,347 @@ private:
+    } else if constexpr (signed_integral<_Ty> && sizeof(_Ty) <= sizeof(long long)) {
+        return static_cast<long long>(42);
+    } else if constexpr (unsigned_integral<_Ty> && sizeof(_Ty) <= sizeof(unsigned int long long)) {
+        return static_cast<unsigned int long long>(42);
+    } else if constexpr (is_same_v<_Ty, bool>) {
+        return bool{};
+    } else if constexpr (is_same_v<_Ty, float>) {
+        return float{};
+    } else if constexpr (is_same_v<_Ty, double>) {
+        return double{};
+    } else if constexpr (is_same_v<_Ty, long double>) {
+        return static_cast<long double>(42);
+    } else if constexpr (is_same_v<_Ty, const void*>) {
+        return static_cast<const void*>(nullptr);
+    } else if constexpr (is_same_v<_Ty, const _CharType*>) {
+        return static_cast<const _CharType*>(nullptr);
+    } else if constexpr (is_same_v<_Ty, basic_string_view<_CharType>>) {
+        return basic_string_view<_CharType>{};
+    } else {
+        return basic_format_arg<_Context>::handle();
+    }
+}
+
+template <class _Context, class _Ty>
+/* consteval */ constexpr size_t _Get_format_arg_storage_size() noexcept {
+    return sizeof(_Get_format_arg_storage_type<_Context, _Ty>());
+}
+
+struct _Format_arg_store_packed_index {
+    // TRANSITION, Should be templated on number of arguments for even less storage
+    using _Index_type = size_t;
+
+    constexpr _Format_arg_store_packed_index() = default;
+    explicit constexpr _Format_arg_store_packed_index(const size_t _Index)
+        : _Index(static_cast<_Index_type>(_Index)), _Type(_Basic_format_arg_type::_None) {}
+
+    _Index_type _Index : (sizeof(_Index_type) * 8 - 4);
+    _Basic_format_arg_type _Type : 4;
 };
 
-// TODO: test coverage
 template <class _Context, class... _Args>
 class _Format_arg_store {
-    static constexpr size_t _Num_args = sizeof...(_Args);
-    // TODO: no arg packing yet
+private:
+    template <class>
+    friend class basic_format_args;
+
+    static constexpr size_t _Num_args       = sizeof...(_Args);
+    static constexpr size_t _Storage_length = (_Get_format_arg_storage_size<_Context, _Args>() + ... + 0);
+
+    using _CharType   = typename _Context::char_type;
+    using _Index_type = _Format_arg_store_packed_index;
+
+    template <class _Ty, _Basic_format_arg_type _Arg_type>
+    void _Store_impl(const size_t _Arg_index, _Ty _Val) noexcept {
+        const auto _Store_index = _Index_array[_Arg_index]._Index;
+        const auto _Length      = _Get_format_arg_storage_size<_Context, _Ty>();
+
+        _CSTD memcpy(_Storage + _Store_index, _STD addressof(_Val), _Length);
+        _Index_array[_Arg_index]._Type = _Arg_type;
+        if (_Arg_index + 1 < _Num_args) { // Set the starting index of the next arg, as that is dynamic
+            _Index_array[_Arg_index + 1] = _Format_arg_store_packed_index{_Store_index + _Length};
+        }
+    }
+
+    byte _Storage[(_STD max)(_Storage_length, static_cast<size_t>(1))];

```

Comment:
Ordinarily, C++20 features can use C++17 features, but `std::byte` has an escape hatch (due to incompatibilities with the Windows SDK). If it's at all possible, `<format>` should not require `byte` internally.